#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

From: Andreas Schwab <schwab@suse.de>
Sender: owner-bash-testers@po.cwru.edu
To: chet@po.cwru.edu, bash-testers@po.cwru.edu
Subject: [bash-testers] $'...' not expanded in rhs of ${name=value}
Date: Fri, 15 Dec 2000 12:19:02 +0100

Configuration Information [Automatically generated, do not change]:
Machine: i386
OS: linux
Compiler: gcc
Compilation CFLAGS:  -DPROGRAM='bash' -DCONF_HOSTTYPE='i386' -DCONF_OSTYPE='linux' -DCONF_MACHTYPE='i386-suse-linux' -DCONF_VENDOR='suse' -DSHELL -DHAVE_CONFIG_H  -D_FILE_OFFSET_BITS=64  -I. -I/usr/include -I. -I./include -I./lib -I/usr/include -O2 -m486 -D_GNU_SOURCE -Wall -pipe -g
uname output: Linux hawking 2.2.16 #1 Wed Aug 2 20:22:26 GMT 2000 i686 unknown
Machine Type: i386-suse-linux

Bash Version: 2.05
Patch Level: 0
Release Status: alpha1

Description:
	In the rhs of an assignment in a parameter expansion the $'...'
	and $"..." expansions are not performed.  This breaks autoconf
	which transforms the assignments in the output of the set command
	from `a=value' into `a={a=value}'.

Repeat-By:
	$ unset a
	$ a=${a=$'foo'}
	$ echo "$a"
	$foo

Fix:
--- bash-2.05-alpha1/variables.c.~1~	Wed Nov  8 18:48:13 2000
+++ bash-2.05-alpha1/variables.c	Fri Dec 15 12:06:49 2000
@@ -852,7 +852,7 @@
     {
       if (quote && sh_contains_shell_metas (var->value))
 	{
-#if 0
+#if 1
 	  t = sh_single_quote (var->value);
 #else
 	  t = ansic_quote (var->value, 0, (int *)0);
